Test/HtmlSupport/QQSpec.hs

```diff
@@ -0,0 +1,72 @@
+{-|
+Module: Test.HtmlSupport.QQSpec
+Copyright: (c) digitally induced GmbH, 2020
+-}
+module Test.HtmlSupport.QQSpec where
+
+import Test.Hspec
+import IHP.Prelude
+import IHP.HtmlSupport.QQ
+import qualified Text.Blaze.Renderer.Text as Blaze
+
+tests = do
+    describe "HSX" do
+        it "should work with static html" do 
```

Comment:
are there any meaningful cases in which we don't want hsx to work but fail in a specific way?

Reply:
Yes, some cases that come to my mind:
- invalid element name (like `<my-invalid-el>`)
- invalid attribute name (like `<div invalid-attribute>`)

We cannot test these cases inside the QQ module as having code such as `[hsx|<my invalid>|]` will fail at compile time. We'd need to add ParserSpec for this that manually runs the HSX parser and then checks the result.

Do you have more than the above two cases in mind?

Reply:
I'm not sure how far you want to push the correctness, but unmatched tags comes to mind

Reply:
Good idea. We should test these as well. This way we can ensure that the parser error messages are good.

Will add this to the PR tomorrow.

Reply:
Added it now :)